### PR TITLE
Small changes, but none to actual code

### PR DESCRIPTION
**Pull request by sokorototo**

### CODE REVIEW

**Comment by sokorototo** [comment]
This commit removes the `vach-assets-manager` crate, adds sorting to the list subcommand in the cli and slims down tge main crate using cargo features.

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [workspace]
 members = [
     "vach",
-    "vach-assets-manager",
     "vach-cli",
     "vach-benchmarks"
 ]
+resolver = "2"
```
Comment (zeskeertwee) [inline Cargo.toml:7]: Why do we need the new resolver?

File: vach/src/lib.rs [hunk 2]
```diff
@@ -4,41 +4,31 @@
 #![deny(missing_docs)]
 
 /*!
-<p align="center">
-  <img src="https://raw.githubusercontent.com/zeskeertwee/virtfs-rs/main/media/logo.png" alt=".vach logo" width="180" height="180">
-</p>
-<p align=center> A simple archiving format, designed for storing assets in compact secure containers </p>
-<p align=center>
-  <a href="https://docs.rs/vach"><img alt="docs.rs" src="https://img.shields.io/docsrs/vach?style=flat-square"></a>
```
Comment (zeskeertwee) [inline vach/src/lib.rs:12]: Why remove all these badges?

File: vach/src/writer/compress_mode.rs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	/// The compressed data is used, only if it is smaller than the original data.
+	Detect,
+	/// The data is never compressed and is embedded as is.
+	Never,
+}
+
+#[cfg(feature = "compression")]
+impl Default for CompressMode {
+	fn default() -> CompressMode {
+		CompressMode::Never
```
Comment (zeskeertwee) [inline vach/src/writer/compress_mode.rs:18]: Wouldn't it make more sense for the default to be Detect?

File: vach/src/writer/leaf.rs [hunk 4]
```diff
@@ -197,11 +187,14 @@ impl<'a> Default for Leaf<'a> {
...
-			#[cfg(feature = "compression")]
-			compress: CompressMode::Never,
+
+			#[cfg(feature = "crypto")]
 			encrypt: false,
+			#[cfg(feature = "crypto")]
 			sign: false,
 
+			#[cfg(feature = "compression")]
+			compress: CompressMode::Never,
```
Comment (zeskeertwee) [inline vach/src/writer/leaf.rs:197]: Same here, wouldn't it make sense to, by default, compress?

**Review by zeskeertwee — CHANGES_REQUESTED** [review]

File: vach-cli/src/main.rs [hunk 5]
```diff
@@ -17,14 +16,10 @@ fn main() {
...
 
-	if matches.is_present(key_names::QUIET) {
-		std::env::set_var("RUST_LOG", "error");
-	};
-
 	let res = match matches.subcommand() {
 		Some((key, mtx)) => commands.get(key).unwrap().evaluate(mtx),
 		None => {
-			println!("vach-cli: Run `vach --help`");
+			println!("vach-cli: Run `vach --help` and refer to crates.io/vach-cli for the manual");
```
[the remark this answers is not in the document]
Comment (sokorototo) [reply vach-cli/src/main.rs:22]: Is there a way to embed man pages into an executable? If not then we will just use include_str!(...) and log the manual

File: vach/src/global/flags.rs [hunk 6]
```diff
@@ -62,7 +62,7 @@ impl Flags {
 	///
 	/// As the [`Flags`] struct uses `u32` under the hood, one can (in practice) set as many as `32` different bits, but some
 	/// are reserved for internal use (ie the first 16 half bits). The good thing is that because of endianness, one can use the remaining 16 bits
-	/// just fine, as seen below. Just using the `0b0000_0000_0000_0000` literal works because `vach` is little endian.
+	/// just fine, as seen below. Just using the `0b0000_0000_0000_0000` literal works because most platforms are little endian.
```
[the remark this answers is not in the document]
Comment (sokorototo) [reply vach/src/global/flags.rs:65]: The `Flags` struct stores a `u32`, therefore is platform dependent. I dont't have a big endian platform to test `vach` on anyway, which is only ARM in a modern context

File: Cargo.toml (hunk 1 above)
Comment (sokorototo) [reply Cargo.toml:7]: Why not

File: vach/src/lib.rs (hunk 2 above)
Comment (sokorototo) [reply vach/src/lib.rs:12]: There are not needed in the API documentation on docs.rs, they are good for crates.io and the repo tho

File: vach/src/writer/leaf.rs (hunk 4 above)
Comment (sokorototo) [reply vach/src/writer/leaf.rs:197]: On it

File: vach-cli/src/main.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (zeskeertwee) [reply vach-cli/src/main.rs:22]: AFAIK the only thing that exists is indeed include_str!

File: vach/src/global/flags.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (zeskeertwee) [reply vach/src/global/flags.rs:65]: Does this mean that it could break on ARM?

File: Cargo.toml (hunk 1 above)
Comment (zeskeertwee) [reply Cargo.toml:7]: But we're already using the 2021 edition, so we've been using the v2 resolver already.

File: vach/src/global/flags.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (sokorototo) [reply vach/src/global/flags.rs:65]: Not really, I especially stated that using that shorthand works in little-endian systems, for big-endian systems use the full `0b0000_0000_0000_0000_0000_0000_0000_0000` literal. Plus it cannot break on ARM `vach` checks for invalid flag insertions.

File: Cargo.toml (hunk 1 above)
Comment (sokorototo) [reply Cargo.toml:7]: Oh ok

**Review by zeskeertwee — APPROVED** [review]

**Comment by sokorototo** [comment]
Aight imma go ahead and publish the new crates

**Pull request merged**